src/lib.rs

```diff
@@ -500,10 +506,31 @@ where
         // Write out the buffer chunks
         for c in buf.chunks(self.chunk_size) {
             self.write_chunk(c)?;
+            self.on_chunk_done();
         }
 
         Ok(())
     }
+
+    /// fadvise that we don't need that data anymore
+    /// fdatassync is required
+    /// https://insights.oetiker.ch/linux/fadvise.html
+    /// fadvise only works on linux, so don't bother is
+    fn on_chunk_done(&mut self) {
```

Comment:
would it be better to just make two versions of this function? One that is `#[cfg(target_os = "linux")]` and another `#[cfg(not(target_os = "linux"))]` that is an empty function?